ext/civiimport/civiimport.php

```diff
@@ -212,7 +212,7 @@ function civiimport_civicrm_alterTemplateFile($formName, $form, $type, &$templat
  *
  * @noinspection PhpUnused
  */
-function civiimport_civicrm_searchKitTasks(array &$tasks, bool $checkPermissions, ?int $userId) {
+function civiimport_civicrm_searchKitTasks(array &$tasks, bool $checkPermissions, ?int $userID, ?array $search = [], ?array $display = []) {
```

Comment:
This is not required for the fix. But it makes sense to update the signature at the same time..

Reply:
@mattwire do we need to allow for NULL - you are adding a default so if not passed it will be `[]` & since these params are new there should be no cases of NULL actively being passed in

Reply:
I don't think so. It should either be unset or an array. The only edge case - but it doesn't get as far as the hook - is the other bit I'm fixing in this PR where the name is passed in as a string before being "expanded" into the array.

Reply:
@mattwire so we can change to just `array $search = []` rather than `?array $search = []`?

Reply:
@eileenmcnaughton Yep, updated. I've rebased this PR so it's just a signature update now

Reply:
Also, turns out is *does* need the `?array` because it can be NULL

Reply:
@mattwire yeah - I just think we should convert it to an array before sending off to the hook - rather than this loosey goosey 'could be this could be that' approach

Reply:
@eileenmcnaughton they way this hook is called from different contexts, there's no way around the fact that sometimes those variables are set and sometimes they are not. However they will always be passed in, even if `NULL`, so @mattwire what I'd like to do with this PR is this:

```suggestion
function civiimport_civicrm_searchKitTasks(array &$tasks, bool $checkPermissions, ?int $userID, ?array $search, ?array $display) {
```